Dockerfile

```diff
@@ -16,6 +16,17 @@ RUN curl -fsSLO "$GORCON_RCONCLI_URL" \
     && rm -Rf "$GORCON_RCONCLI_DIR" \
     && go build -v ./cmd/gorcon
 
+
+WORKDIR /build/custom_rcon_broadcast/
+
+# Download the source code
+RUN curl -OJL https://raw.githubusercontent.com/thejcpalma/palworld-dedicated-server-docker/0.2.1/src/custom_rcon_broadcast/go.mod \
+    && curl -OJL https://raw.githubusercontent.com/thejcpalma/palworld-dedicated-server-docker/0.2.1/src/custom_rcon_broadcast/go.sum \
+    && curl -OJL https://raw.githubusercontent.com/thejcpalma/palworld-dedicated-server-docker/0.2.1/src/custom_rcon_broadcast/main.go
```

Comment:
Wouldn't it make more sense to make a repository just for the RCON broadcast script, and reference that, instead of it being within the tree of another repo?

Reply:
I think you both mean the same 🤣 
I also think when i said that it was about a PR with something about a Python web-ui or something like that, which was 1) out of scope for my old vision (minor point) of this image and 2) it was Python, i dont know Python i cant fix it and support it. Thats why i wanted to not add that and im sure i said something around the lines what @thejcpalma said.

For me that falls under sw-arch and seperation of concerns, you juggle with it and figure out which one you need more and are happy with.

Imho, i have seen many Docker-Topic-Projects who do the same as @thejcpalma does but also many that do what @Callum027 says. Its both out there plenty of times and im sure everything has their reason for doing what they do, or they wouldnt do it. I personally find it easier in a "seperate" repo, which i then can handle/treat/distinguish as a "vendor" like i can in PHP, so basically a component or library i want to use, does only partly matter if i have to build it myself or not to be honest.

BUT one thing i have to give @Callum027 - That code doesnt follow Best-Practices or the Ansible-Way with version-pinning and checking every asset for "hash" if you didnt suffer a MITM attack. Example here: https://github.com/jammsen/docker-palworld-dedicated-server/blob/develop/Dockerfile#L5 Line 5-32 should be plenty to explain this.

Reply:
Doesn't make sense to hash source code. Hashing source code files is not a common practice in the same way it is for binaries or other distributable artifacts/packages/assets. If source code files are version-controlled their integrity is maintained through version history and GitHub.

When you distribute a compiled binary or an executable, it's common to provide a hash value (like the MD5 on the gorcon's rcon-cli) alongside the binary. This hash allows us to verify the integrity of the downloaded binary to ensure it hasn't been tampered with during the download process.

When users clone or pull the repository, they get the entire version history and can trust the source code they receive, assuming they are interacting with a legitimate repository.

It's worse the case with having `gosu-amd64` binary directly on the repository, not knowing if it was tampered with. Even on gosu github repo they only have checksums for the binaries and the signature files of those binaries.
I would be better to get the files from the repo and rebuild with golang version 1.22 on the dockerfile than having a binary on the repo (like you always say, you don't want bitcoin miners 😅) which might be harmful because there is no transparency.

Reply:
> Doesn't make sense to hash source code. Hashing source code files is not a common practice in the same way it is for binaries or other distributable artifacts/packages/assets. If source code files are version-controlled their integrity is maintained through version history and GitHub.

Depends on the way to getting that code. 
 
> It's worse the case with having `gosu-amd64` binary directly on the repository, not knowing if it was tampered with. Even on gosu github repo they only have checksums for the binaries and the signature files of those binaries. I would be better to get the files from the repo and rebuild with golang version 1.22 on the dockerfile than having a binary on the repo (like you always say, you don't want bitcoin miners 😅) which might be harmful because there is no transparency.

Huh? What? Assuming and knowing is different. I did talk to a rl-friend of mine who does a lot of go, he helped me to undestand and fix the problems, with the many CVEs, then we downloaded the source, made the needed changes, then i build the binary myself, scanned it with 3 tools for CVEs (Synk, Sonatype and Docker-Scout) after that we finished it by using virustotal.com scan and then i added it to the repo, to make sure we have a running binary, thats not depended on the repo-maintainer, to fix CVEs. Neither the rcon-cli nor gosu maintainer seem to be active at the moment.

Reply:
> Huh? What? Assuming and knowing is different. I did talk to a rl-friend of mine who does a lot of go, he helped me to undestand and fix the problems, with the many CVEs, then we downloaded the source, made the needed changes, then i build the binary myself, scanned it with 3 tools for CVEs (Synk, Sonatype and Docker-Scout) after that we finished it by using virustotal.com scan and then i added it to the repo, to make sure we have a running binary, thats not depended on the repo-maintainer, to fix CVEs. Neither the rcon-cli nor gosu maintainer seem to be active at the moment.

You're taking this in a personal way. I simply mean that having the source code you used, changed, scans you did, conveys transparency. If someone just looks at the repo and sees a binary without any more info, they'll get suspicious (again, I'm not doubting you in any way).

I have this in my repo with a folder for this custom binaries (currently just one but if anymore are needed) and their respective source code. That way, people can use the feature and have access to the source code and know for a fact and without a single doubt what that code is doing because it's there. You can also copy the code on my repo here. I just pinned it to a release where it's working and was curl'ing it on your dockerfile since I'm the one maintaining it and if changes are needed I'll adapt it in the repo, and you just need to change the version.